### PR TITLE
[IT-1655] Add caching layer to MIPS microservice

### PR DESCRIPTION
**Pull request by jesusaurus**
Cache results in S3 to avoid frequent interaction with the upstream API and to provide some resiliency.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,17 +22,32 @@ The following template parameters are set as environment variables in the lambda
...
+```
+#### Respones Format
 
 The API will return a json string representing a dictionary mapping program codes to their names.
 E.g.:
 ```json
 {"000000": "No Program", "990300": "Program Infrastructure"}
 ```
 
+### Cache Administration
```
Comment (zaro0508) [inline README.md:42]: can you please create a section to explain the purpose for creating a cache for this simple service?  

Also are you planning to integrate the nice new caching images into the docs somewhere?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
I don't think you need to reinvent the wheel for caching, there are already existing python and lambda libraries that you can use, please take a look at https://blog.devgenius.io/aws-lambda-caching-why-and-how-bd3c2d500c

**Comment by zaro0508** [comment]
Lambda deploys on the API gateway @jesusaurus  and it looks like the API gateway already has a cache you can just use.  Look at the docs https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-caching.html

**Comment by jesusaurus** [comment]
> The maximum TTL value is 3600 seconds

The API Gateway cache has a maximum TTL of one hour, and I don't think that'll be very useful for us since we're not likely to be hitting the API every hour. For our use case, we want to specify our TTL in number of hours or days.

**Comment by zaro0508** [comment]
Still waiting for the explanation of this use case and why you need caching.  If you need something with longer cache expiration then how about aws elasticache?  you can set expiration to greater than 30 days.  https://www.tutorialspoint.com/memcached/memcached_set_data.htm

```
exptime − It is the expiration time in seconds. 0 means no delay. If exptime is more than 30 days, Memcached uses it as UNIX timestamp for expiration.
```

**Comment by jesusaurus** [comment]
@zaro0508 can you explain what's wrong with the current implementation? What makes building an implementation on top of elasticache more apporiate than the current solution using s3?

**Comment by jesusaurus** [comment]
I just realized I've been focusing on implementing a cache inside the lambda, when we could use cloudfront as a cache in front of the lambda -- this is worth evaluating

**Comment by jesusaurus** [comment]
closed in favor of #4

**Pull request closed without merging**